### ISSUE TITLE
Value cannot be null.:I2C Write Exception: {0}

### ISSUE THREAD

**Issue by FishOfPrey** [adafruit/AdafruitClassLibrary#6]
When attempting to use the nuget packages PCA9685 class I was getting the following error logged after calling `SetAllPWM()`.

**Value cannot be null.:I2C Write Exception:**

Instead of the nuget package I pulled the latest source down from here. The null reference exception was coming from Device being null. At first I thought it wasn't being set correctly in InitI2CAsync, but then found that InitPCA9685Async was also throwing and logging and exception. With the exceptions being logged and then ignored it's difficult to go straight to the cause of the problem.

Then underlying error was:

**Could not load file or assembly 'System.Runtime.WindowsRuntime,...**

Found the [solution](https://github.com/KSemenenko/ColorThief/issues/3#issuecomment-251656380):

> the default NuGet package (Microsoft.NETCore.UniversalWindowsPlatform) had to be upgraded to the latest version (5.2.2) as well!

Once I updated UWP to 5.2.3 it started working.

**Comment by lloydjatkinson** [comment]
I agree, logging the exception and then immediately discarding it is not the approach that should be taken, but it currently is in the method you mentioned:

        public async Task InitPCA9685Async(I2CSpeed i2cSpeed = I2CSpeed.I2C_100kHz)
        {
            try
            {
                await InitI2CAsync(i2cSpeed);

                Reset();
            }
            catch (Exception e)
            {
                System.Diagnostics.Debug.WriteLine("Exception: {0}", e.Message);
                return;
            }
        }

Especially not in an "init" type method, it should either succeed or not and allow the developer to decide what happens when it does not succeed.

**Comment by SindreSB** [comment]
I just had this same issue. The project needs to be updated to UWP 5.4.0. 

Solutuon was to download the source, and update it.
